marlowe-playground-client/src/Halogen/BlocklyCommons.purs

```diff
@@ -0,0 +1,71 @@
+-- TODO In a different commit I'd rename Blockly to BlocklyEditor and ActusBlockly as ActusEditor or
+-- ActusBlocklyEditor and this module could be called solely Blockly
+module Halogen.BlocklyCommons where
+
+import Blockly (addChangeListener, removeChangeListener)
+import Blockly.Events (fromEvent, newParentId, oldParentId)
+import Blockly.Types (BlocklyEvent, Workspace)
+import Blockly.Types as BT
+import Control.Alt ((<|>))
+import Data.Foldable (oneOf)
+import Data.Lens (Lens', assign)
+import Data.Lens.Record (prop)
+import Data.Symbol (SProxy(..))
+import Data.Traversable (for_)
+import Effect.Aff.Class (class MonadAff)
+import Halogen (HalogenM, raise)
+import Halogen.Query.EventSource (EventSource)
+import Halogen.Query.EventSource as EventSource
+import Prelude (Unit, bind, discard, pure, unit, ($), (<$>))
+import Web.Event.EventTarget (eventListener)
+
+blocklyEvents ::
+  forall m action.
+  MonadAff m =>
+  (BlocklyEvent -> action) ->
+  Workspace ->
+  EventSource m action
+blocklyEvents toAction workspace =
+  EventSource.effectEventSource \emitter -> do
+    listener <-
+      eventListener \event -> do
+        let
+          mEvent =
+            -- Blockly can fire all of the following events https://developers.google.com/blockly/guides/configure/web/events
+            -- but at the moment we only care for the ones that may affect the unsaved changes.
+            oneOf
+              [ BT.Create <$> fromEvent event
+              , BT.Move <$> fromEvent event
+              , BT.Change <$> fromEvent event
+              , BT.FinishLoading <$> fromEvent event
+              ]
```

Comment:
Here the `do` is not necessary. Just have to add `in` after the `let`. Same amount of characters, but it is a bit simpler, I think.

Reply:
done